Libraries/graphics/Headers/graphics/PixelFormat.h

```diff
@@ -0,0 +1,149 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#ifndef __graphics_PixelFormat_h
+#define __graphics_PixelFormat_h
+
+#include <vector>
+
+#include <cstddef>
+#include <cstdint>
+
+namespace graphics {
+
+/*
+ * Describes the format of pixels in a bitmap. Assumes 8 bits per channel.
+ */
+class PixelFormat {
+public:
+    /*
+     * The channels stored in the bitmap.
+     */
+    enum class Color {
+        /*
+         * Only a gray channel.
+         */
+        Grayscale,
+        /*
+         * RGB color.
+         */
+        RGB,
+    };
+
+    /*
+     * The order of the channels.
+     */
+    enum class Order {
+        /*
+         * Forward order in memory, i.e. big-endian.
+         */
+        Forward,
+        /*
+         * Reversed order in memory, i.e. little-endian.
+         */
+        Reversed,
```

Comment:
Listing out all the possible formats explicitly is impractical, since this class supports `2 * 2 * 7 = 28` formats already and I can imagine many more. That would also mean additional functions to get out the information here as it's much easier to switch on "RGB or Gray" than "RGB, BGR, RGBA, ARGB, BGRA, etc vs GA, G, AG, etc".

CoreGraphics uses "big endian" and "little endian" for this, which helps in that it's obviously referring to the channels, but I find it confusing in that it isn't clear what the base value is that the endianness modifies. This makes it a little more clear in that RGB is the base that this can change the ordering of.

I think the important part is that the pixel format only talks about the format of a specific pixel in the image — not the relationship *between* pixels. So being nested in that makes it more clear that it's just for one pixel.

I'm open to alternatives here on naming, though. This isn't my favorite, just the best I came up with.